subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompilerTest.kt

```diff
@@ -471,7 +485,8 @@ class ResidualProgramCompilerTest : TestWithTempFiles() {
                     scriptTemplateId,
                     sourceHash,
                     ProgramKind.TopLevel,
-                    ProgramTarget.Project)
+                    ProgramTarget.Project,
+                    accessorsClassPath)
```

Comment:
This method now has 9 parameters. It might be reaching the point where it would make sense to group them into an object.

Reply:
Generally I'd agree with you, @JLLeitschuh, in this concrete case however, that would mean memory indirections which is exactly the sort of thing I'm trying to avoid.

Reply:
I didn't realize that this code was that much of a "hot path" that you were trying to optimize for.
Fair enough.
👍